lumibot/strategies/strategy_executor.py

```diff
@@ -165,38 +165,35 @@ def func_output(self, *args, **kwargs):
 
             # ORDERS
             if len(orders_broker) > 0:
-                orders_lumi = self.broker._tracked_orders
+                orders_lumi = self.broker.get_all_orders()
 
                 # Check orders at the broker against those in lumibot.
                 for order in orders_broker:
-                    if self.strategy._first_iteration:
-                        self.broker._process_new_order(order)
+                    # Check against existing orders.
+                    order_lumi = [ord_lumi for ord_lumi in orders_lumi if ord_lumi.identifier == order.identifier]
+                    order_lumi = order_lumi[0] if len(order_lumi) > 0 else None
+
+                    if order_lumi:
+                        # Compare the orders.
```

Comment:
Don't we already have code for this in the Order class? I think you just need to check if the two orders are equal